apps/src/templates/teacherNavigation/ElementOrEmptyPage.tsx

```diff
@@ -27,82 +31,159 @@ const ElementOrEmptyPage: React.FC<ElementOrEmptyPageProps> = ({
   showNoStudents,
   showNoCurriculumAssigned,
   showNoUnitAssigned,
+  showNoLessonMaterialsForLegacyCourses,
+  showNoCalendarForLegacyCourses,
+  showNoLessonMaterialsForLesson,
+  showNoCalendarForThisUnit,
+  isOnCalendarPage,
   courseName,
   element,
 }) => {
   const isLoadingSectionData = useAppSelector(
     state => state.teacherSections.isLoadingSectionData
   );
 
-  const textDescription = () => {
-    if (showNoStudents) {
-      return i18n.emptySectionDescription();
-    } else if (showNoCurriculumAssigned) {
-      return i18n.noCurriculumAssigned();
-    } else {
-      return i18n.noUnitAssigned({courseName: courseName});
-    }
-  };
+  const lessonMaterialsOrCalendarPage = isOnCalendarPage
```

Comment:
Not 100% sure on this approach.  We need someway to indicate which page we are on for the strings in the empty state messages but also to decide which graphics to show.  Open to alternatives here.

Reply:
Could you use the URL path?